scanomatic/models/factories/analysis_factories.py

```diff
@@ -142,167 +142,144 @@ def all_keys_valid(cls, keys):
-        """
+    def _validate_pinning_matrices(cls, model: analysis_model.AnalysisModel):
         if AbstractModelFactory._is_pinning_formats(model.pinning_matrices):
             return True
         return model.FIELD_TYPES.pinning_matrices
 
     @classmethod
-    def _validate_use_local_fixture(cls, model):
-        """
-
-        :type model: scanomatic.models.analysis_model.AnalysisModel
-        """
+    def _validate_use_local_fixture(cls, model: analysis_model.AnalysisModel):
         if isinstance(model.use_local_fixture, bool):
             return True
         return model.FIELD_TYPES.use_local_fixture
 
     @classmethod
-    def _validate_stop_at_image(cls, model):
-        """
-
-        :type model: scanomatic.models.analysis_model.AnalysisModel
-        """
+    def _validate_stop_at_image(cls, model: analysis_model.AnalysisModel):
         if isinstance(model.stop_at_image, int):
             return True
         return model.FIELD_TYPES.stop_at_image
 
     @classmethod
-    def _validate_output_directory(cls, model):
-        """
-
-        :type model: scanomatic.models.analysis_model.AnalysisModel
-        """
-        if (model.output_directory is None or
-                isinstance(model.output_directory, StringTypes) and
-                os.sep not in model.output_directory):
+    def _validate_output_directory(cls, model: analysis_model.AnalysisModel):
+        if (
+            model.output_directory is None
+            or isinstance(model.output_directory, StringTypes)
+            and os.sep not in model.output_directory
+        ):
             return True
         return model.FIELD_TYPES.output_directory
 
     @classmethod
-    def _validate_focus_position(cls, model):
-        """
-
-        :type model: scanomatic.models.analysis_model.AnalysisModel
-        """
+    def _validate_focus_position(cls, model: analysis_model.AnalysisModel):
         if model.focus_position is None:
             return True
 
         is_coordinate = (
-            cls._is_tuple_or_list(model.focus_position) and
-            all(isinstance(value, int) for value in model.focus_position) and
-            len(model.focus_position) == 3)
+            cls._is_tuple_or_list(model.focus_position)
+            and all(isinstance(value, int) for value in model.focus_position)
+            and len(model.focus_position) == 3
+        )
 
         if is_coordinate and cls._validate_pinning_matrices(model):
 
             plate_exists = (
-                0 <= model.focus_position[0] < len(model.pinning_matrices) and
-                model.pinning_matrices[model.focus_position[0]] is not None)
-
-            if plate_exists and (
-                    0 <= val < dim_max for val, dim_max in
-                    zip(
+                0 <= model.focus_position[0] < len(model.pinning_matrices)
+                and model.pinning_matrices[model.focus_position[0]] is not None
+            )
+
+            if (
+                plate_exists and all(
```

Comment:
added `all` here because:

1. A generator is always truthy (only had generator before)
2. `_validate_grid_images` a bit further down uses similar logic and has `all`.